### ISSUE TITLE
Add support of TS type annotations on `ofetch.create`

### ISSUE THREAD

**Issue by ysemennikov** [unjs/ofetch#406]
### Describe the feature

Hi everyone!

I really love using ofetch because it's very powerful and covers almost all use cases.

The one feature that is missing for me is type annotatations. I believe this could be automated using tools like [openapi-typescript](https://github.com/drwpow/openapi-typescript), which can generate TS-types based on OpenAPI specification.

So what I mean is:
```ts
// import types auto-generated from the OpenAPI spec
// they're generated by openapi-typescript
import type { paths } from './autogenerated-types/my-api';

const myApiFetch = ofetch.create<paths>(options)  // provide paths as generic to ofetch.create

// intellisense for request URL, method, body, etc.
myApiFetch('/endpoint', {
  method: 'POST',
  body: {}
})
```

The `openapi-typescript` has its own package called `openapi-fetch` with support of this feature, but since `ofetch` is great integrated into Nuxt 3, it would be very nice to also have this possibility here. And `ofetch` has more features than the `openapi-fetch`.

### Additional information

- [X] Would you be willing to help implement this feature?

**Comment by TimGuendel** [comment]
+1 This would be really nice

**Comment by martin-juul** [comment]
bumping this since i'm running into the same issue.

**Comment by martin-juul** [comment]
The typing issue doesn't seem to appear if you use the library like this:

```
export function createLibrary(data: CreateLibraryRequest) {
  return apiFetch<Library>(route('api.library.create'), {
    method: 'POST',
    body: data,
  })
}
```

I ran into the same issue like you, but if i adjust my request all seems to be good.

**Comment by ysemennikov** [comment]
> The typing issue doesn't seem to appear if you use the library like this:
> 
> ```
> export function createLibrary(data: CreateLibraryRequest) {
>   return apiFetch<Library>(route('api.library.create'), {
>     method: 'POST',
>     body: data,
>   })
> }
> ```
> 
> I ran into the same issue like you, but if i adjust my request all seems to be good.

You are right, but in this case you need to manually define a return type for each request you send. 

I suggest another feature - creating an ofetch instance and passing a generic type like this: 
`const apiFetch = ofetch.create<apiTypes>()`.

And then all requests will be already properly typed.

**Comment by MuhammadM1998** [comment]
Is there currently any workaround to this?

I'm using [Nuxt API Party](https://nuxt-api-party.byjohann.dev) which gives me a fetch client (based on [openapi-typescript](https://github.com/drwpow/openapi-typescript)). I'm passing an openapi schema so I'm getting type-safety for paths, queries, etc.

I need custom interceptors logic so I tried passing the generated fetch client to `createFetch` function like below but I lose the type hints
```ts
// $typedFetch is the fetch client generated by API Party module
const customFetch = createFetch({ fetch: $typedFetch }).create({
  baseURL: 'my-api-url',
  onRequest({ options }) {
    // Custom options
  },
})

// This doesn't provide intellisense
await customFetch('endpoint-path')

// This does provide intelliSense, but can't extend it
await $typedFetch('endpoint-path')
```
